modules/test-utils/src/snapshot-test-runner.ts

```diff
@@ -40,77 +71,78 @@ const DEFAULT_TEST_CASE = {
   goldenImage: ''
 };
 
-export default class SnapshotTestRunner extends TestRunner {
-  constructor(props) {
-    super(props);
+export class SnapshotTestRunner extends TestRunner<
+  SnapshotTestCase,
+  DiffImageResult,
+  {imageDiffOptions: ImageDiffOptions}
+> {
+  private _isDiffing: boolean = false;
 
-    this.isDiffing = false;
+  constructor(props) {
+    super(props, DEFAULT_TEST_OPTIONS);
 
-    Object.assign(this.testOptions, DEFAULT_TEST_OPTIONS);
+    this._isDiffing = false;
   }
 
   get defaultTestCase() {
     return DEFAULT_TEST_CASE;
   }
 
-  initTestCase(testCase) {
+  initTestCase(testCase: SnapshotTestCase) {
     super.initTestCase(testCase);
     if (!testCase.goldenImage) {
       throw new Error(`Test case ${testCase.name} does not have golden image`);
     }
   }
 
-  runTestCase(testCase, onDone) {
-    const {deck} = this;
-
-    deck.setProps({
-      ...this.props,
-      ...testCase,
-      onBeforeRender: () => {
-        testCase.onBeforeRender({
-          deck,
-          layers: deck.props.layers
-        });
-      },
-      onAfterRender: () => {
-        testCase.onAfterRender({
-          deck,
-          layers: deck.props.layers,
-          done: onDone
-        });
-      }
-    });
-  }
+  runTestCase(testCase: SnapshotTestCase) {
```

Comment:
Any reason not to use `async` like [here](https://github.com/visgl/deck.gl/pull/8483/files#diff-692be60939fe8db8093274603fe97b770643e81688e9c91404107d1e93d241e7R60)?

Reply:
deck's render event is callback based.